### DISCUSSION TITLE
useSearchParams not working in HashRouter

### DISCUSSION THREAD

**Discussion by MuhammadAnas190** [remix-run/react-router#10274] [Q&A]
I would like to know if this is the default behaviour for `useSearchParams` to only operate in `<BrowserRouter>`.
Currently `useSearchParams` returns an empty object on `<HashRouter>`.
Please check the attached screenshot
<img width="602" alt="Screenshot 2023-03-28 at 2 11 28 PM" src="https://user-images.githubusercontent.com/42142889/228189302-48b185dd-f57c-4ede-b864-dba265567f03.png">

Code implementation:
```
- App.js
<MemoryRouter>
  {
    useRoutes({
    path: "/cms",
    element: <BaseLayout  />,
    children: [
      **Private and public app routes**
      {
        path: "*",
        element: (
          <Result
            status="404"
            title="404"
            subTitle="Sorry, the page you visited does not exist."
          />
        ),
      },
    ],
  },)
  }
</MemoryRouter>
```
```
- PrivateFile.js
  const [searchParams] = useSearchParams();
  console.log("🚀 ~ file: authorization.js:19 ~ searchParams:", searchParams);
```

**Comment by awais-100** [comment]
same issue here, but it's working when I put the "#" symbol before "?" in url
